### PR TITLE
feat(bigquery): add arrow row format example

### PR DESCRIPTION
**Pull request by alvarowolfx**
Improve BigQuery quickstart by adding an example on how to use the Arrow format.

### CODE REVIEW

**Review by shollyman — APPROVED** [review]
Nice!  It looks like the work on arrow flight is where the necessary changes in the schema handling in arrow/ipc landed.  Thanks for taking a look at this!

**Comment by alvarowolfx** [comment]
Resolves https://github.com/GoogleCloudPlatform/golang-samples/issues/1946

File: bigquery/bigquery_storage_quickstart/main.go [hunk 1]
```diff
@@ -266,7 +314,49 @@ func processStream(ctx context.Context, client *bqStorage.BigQueryReadClient, st
...
+	r, err := ipc.NewReader(buf, ipc.WithAllocator(mem))
+	if err != nil {
+		return err
+	}
+	aschema := r.Schema()
+	for {
+		select {
+		case <-ctx.Done():
+			// Context was cancelled.  Stop.
+			return nil
```
Comment (codyoss) [inline bigquery/bigquery_storage_quickstart/main.go:339]: nit: This error should be propagated back to the user.

Comment (alvarowolfx) [reply bigquery/bigquery_storage_quickstart/main.go:339]: Does returning `ctx.Err()` make sense in this context ? 

```
case <-ctx.Done():
	// Context was cancelled.  Stop.
	return ctx.Err()
```

**Review by codyoss — APPROVED** [review]

**Pull request merged**